test/wasserstein.jl

```diff
@@ -0,0 +1,81 @@
+using OptimalTransport
+
+using Distances
+using Distributions
+
+using Random
+using Test
+
+Random.seed!(100)
+
+@testset "wasserstein.jl" begin
+    @testset "p2distance" begin
+        for metric in (Euclidean(), Euclidean(0.01), TotalVariation())
+            @test OptimalTransport.p2distance(metric, Val(1)) === metric
+        end
+
+        @test OptimalTransport.p2distance(Euclidean(), Val(2)) == SqEuclidean()
+        @test OptimalTransport.p2distance(Euclidean(0.01), Val(2)) == SqEuclidean(0.01)
+
+        p = randexp()
+        x = randn(10)
+        y = randn(10)
+        for metric in (Euclidean(), TotalVariation())
+            pmetric = OptimalTransport.p2distance(metric, Val(p))
+            @test pmetric(x, y) ≈ metric(x, y)^p
+        end
+    end
+
+    @testset "prt" begin
+        x = randexp()
+        for p in (1, 2, 3, randexp())
+            @test OptimalTransport.prt(x, Val(p)) ≈ x^(1 / p)
+        end
+    end
+
+    @testset "wasserstein" begin
+        μ = Normal(randn(), randexp())
+        ν = Normal(randn(), randexp())
+        for p in (1, 2, 3, randexp()), metric in (Euclidean(), TotalVariation())
+            # without additional keyword arguments
+            w = wasserstein(μ, ν; p=Val(p), metric=metric)
```

Comment:
is it necessary to use `p=Val(p)` when using the `wasserstein` function?

Reply:
Yes, otherwise one can't dispatch on `p` to use optimized cost functions such as `SqEuclidean`.

Reply:
Hmm. I understand, although it's a bit unfortunate, cause it's kind of odd.

Reply:
In the latest commit I added support for `Real`-valued `p`s and a note in the docstring that explains that sometimes `Val(p)` might be more efficient.